app/inbound/d365/process-d365.js

```diff
@@ -0,0 +1,37 @@
+const db = require('../../data')
+const saveD365 = require('./save-d365')
+const validateD365 = require('./validate-d365')
+const getD365ByPaymentReference = require('./get-d365-by-payment-reference')
+
+const processD365 = async (d365) => {
+  const transaction = await db.sequelize.transaction()
+
+  try {
+    const existingD365 = await getD365ByPaymentReference(d365.paymentReference, transaction)
+    if (existingD365) {
+      console.info(`Duplicate D365 paymentReference received, skipping ${existingD365.paymentReference}`)
+      await transaction.rollback()
+    } else {
+      // De-aliasing calculationReference
+      const transformedD365 = {
+        ...d365,
+        calculationId: d365.calculationReference
```

Comment:
oh so we need to have calculationReference here instead of id? I'm really struggling to see why we use this workaround...

Reply:
Yes, and I agree, I do not understand why we have a restriction in data to stop us using Id and creating these aliases being required. Anything ending in Id must be changed upstream but from here onward that mapping isn't required anywhere.